### PR TITLE
[Component with its own class name as service type] LPS-193631 Refactor ModelResourcePermission components

### PR DESCRIPTION
**Pull request by MrJohn1911**
Ticket: https://liferay.atlassian.net/browse/LPS-193631

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:data-engine

**Comment by MrJohn1911** [comment]
Last push fixes conflicts with master.

**Comment by catarineaguiar** [comment]
no regressions found

**Comment by catarineaguiar** [comment]
ci:test:sf

File: modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/resource/v2_0/DataListViewResourceImpl.java [hunk 1]
```diff
@@ -319,8 +322,20 @@ private long _getClassNameId() {
 		return _portal.getClassNameId(DEDataListView.class);
 	}
 
-	private long _getDDMStructureId(DEDataListView deDataListView) {
-		return deDataListView.getDdmStructureId();
+	private DDMStructure _getDDMStructure(DEDataListView deDataListView) {
```
Comment (shuyangzhou) [inline modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/resource/v2_0/DataListViewResourceImpl.java:325]: @MrJohn1911 this is not ok. In the original logic the ddmStructureId was passed into the DataDefinitionModelResourcePermission.check() which internally did a DDMStructureLocalService.getDDMStructure() call. On non-existing id, it throws PortalException. But after your change it passes down a null to the new check method which internal just use the reference directly which will lead to a NPE.

**Comment by shuyangzhou** [comment]
@MrJohn1911 @rafaprax See above comment, please fix and resend.

**Pull request closed without merging**